cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -76,14 +78,14 @@ def valid_path?(path)
   return false if path.length > 255
   # Valid characters allowed
   char = /[A-Za-z0-9_\-.$\/~"'@:+]/
-  # Exactly one wildcard required at start or end
-  return false unless path.match /^(\*#{char}*|#{char}*\*)$/
+  # Maximum one wildcard allowed at start or end
+  return false unless path.match /^(\*#{char}*|#{char}*\*|#{char}*)$/
```

Comment:
This is hard to read due to the mix of escaped and non-escaped stars.  How about this:

chars = /[A-Za-z0-9_\-.$\/~"'@:+]*/
path.match /^ (\*#{chars} | #{chars}\* | #{chars}) $/x.present?

(We use /x to allow whitespace in the regexp.  .present? allows us to avoid the awkward if.)